### PR TITLE
Disable usage of `var`.

### PR DESCRIPTION
**Pull request by rwjblue**
* Prefer `const` for `require`'s
* Prefer `let` for everything else.

### CODE REVIEW

File: blueprints/addon/index.js [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
+
+const fs = require('fs-extra');
+const existsSync = require('exists-sync');
+const path = require('path');
+const walkSync = require('walk-sync');
+const stringUtil = require('ember-cli-string-utils');
+const uniq = require('ember-cli-lodash-subset').uniq;
+const SilentError = require('silent-error');
+const sortPackageJson = require('sort-package-json');
+let date = new Date();
```
Comment (stefanpenner) [inline blueprints/addon/index.js]: i would expect this also to be `const` does the linting prevent this?

Comment (rwjblue) [inline blueprints/addon/index.js]: @stefanpenner - The linting rules absolutely still allow us to use `const` there, but I used the basic heuristic of "if its a `require` then make it `const`" to automate these changes and this one just ended up being a `let`.

Once we land this we should start reviewing "whats left" and fixing things like this one...

Comment (stefanpenner) [inline blueprints/addon/index.js]: sounds great.

**Comment by stefanpenner** [comment]
We shouldn't let this sit for long, but lets make sure we coordinate with @nathanhammond re: landing this in a way thats aligned with his plan here. (He thought about it thoroughly, don't want to mis some aspect of his plan) If that is already done, :shipit:

**Comment by rwjblue** [comment]
@stefanpenner - Yep, @Turbo87 @nathanhammond and I circled the wagons last night and are on the same page.  We basically want to get all of this "large scale refactoring" done and out of the way before branch point into 2.12's beta cycle.

**Comment by rwjblue** [comment]
@homu r+

**Pull request merged**